test/unit/test_grouped_loader.py

```diff
@@ -0,0 +1,238 @@
+                    data_dir_path=data_dir_path, image_shape=image_shape, **common_args
+                )
+
+                assert data_loader.validate_data_files() is None
+
+
+def test_num_intra_group_probs():
+    """
+    Test whether fewer than two groups throws an error for intra_group_prob < 1
+     - data/test/h5/grouped/test contains only a single group
+    """
+    with pytest.raises(Exception) as e_info:
+        data_dir_path = join(DataPaths["h5"], "test")
+        image_shape = (64, 64, 60)
+        common_args = dict(
+            file_loader=H5FileLoader,
+            labeled=True,
+            sample_label="all",
+            intra_group_prob=0.5,
+            intra_group_option="forward",
+            sample_image_in_group=False,
+            seed=0,
+        )
+        data_loader = GroupedDataLoader(
+            data_dir_path=data_dir_path, image_shape=image_shape, **common_args
+        )
+        data_loader.close()
+
+        assert "we need at least two groups" in str(e_info.value)
+
+
+def test_sample_indices_intra():
+    """
+    Test the number of samples and indices are correct for a single image sample
+    """
+    data_dir_path = join(DataPaths["h5"], "train")
+    image_shape = (64, 64, 60)
+    common_args = dict(
+        file_loader=FileLoaderDict["h5"],
+        labeled=True,
+        sample_label="all",
+        intra_group_prob=0.5,
+        intra_group_option="forward",
+        sample_image_in_group=True,
+        seed=None,
+    )
+    data_loader = GroupedDataLoader(
+        data_dir_path=data_dir_path, image_shape=image_shape, **common_args
+    )
+
+    assert data_loader.sample_indices is None
+    assert data_loader._num_samples == 2
+
+
+def test_group_indices_inter():
+    """
+    Test the number of samples and indices are correct when entire group is sampled
+    """
+    data_dir_path = join(DataPaths["h5"], "train")
+    image_shape = (64, 64, 60)
+    common_args = dict(
+        file_loader=FileLoaderDict["h5"],
+        labeled=True,
+        sample_label="all",
+        intra_group_prob=0,
+        intra_group_option="forward",
+        sample_image_in_group=False,
+        seed=None,
+    )
+    data_loader = GroupedDataLoader(
+        data_dir_path=data_dir_path, image_shape=image_shape, **common_args
+    )
+
+    ni = np.array(data_loader.num_images_per_group)
+    num_samples = np.sum(ni) * (np.sum(ni) - 1) - sum(ni * (ni - 1))
+
+    sample_indices = data_loader.sample_indices
+    sample_indices.sort()
+    unique_indices = list(set(sample_indices))
+    unique_indices.sort()
+
+    assert data_loader._num_samples == num_samples
+    assert sample_indices == unique_indices
+
+
+def test_group_indices_intra():
+    """
+    Test sample numbers and indices are correct when entire group is sampled
+    """
+    data_dir_path = join(DataPaths["h5"], "train")
+    image_shape = (64, 64, 60)
+    for group_option in ["forward", "backward", "unconstrained"]:
```

Comment:
If you check https://codecov.io/gh/DeepRegNet/DeepReg/pull/247/src/deepreg/dataset/loader/grouped_loader.py?before=deepreg/dataset/loader/grouped_loader.py, we will find line 147 not covered,

so we need to test a not supported group_option like "not_supported", and check the error

Reply:
Thanks, I missed that one! Amended